corehq/apps/saved_reports/tasks.py

```diff
@@ -198,3 +174,48 @@ def send_email_report(self, recipient_emails, domain, report_slug, report_type,
             export_all_rows_task(config.report, report_state, recipient_list=recipient_emails)
         else:
             self.retry(exc=er)
+
+
+def create_config_for_email(report_type, report_slug, user_id, domain, request_data):
+    config = ReportConfig()
+
+    # see ReportConfig.query_string()
+    object.__setattr__(config, '_id', 'dummy')
+    config.name = _("Emailed report")
+    config.report_type = report_type
+    config.report_slug = report_slug
+    config.owner_id = user_id
+    config.domain = domain
+
+    GET = dict(six.iterlists(request_data['GET']))
+    exclude = ['startdate', 'enddate', 'subject', 'send_to_owner', 'notes', 'recipient_emails']
+    filters = {}
+    for field in GET:
+        if field == 'params':
+            params = unquote(GET.get(field)[0])
+            params = params.split('&')
+            for param in params:
+                field, value = tuple(param.split('=', 1))
```

Comment:
Reusing the name `field` here seems dangerous. I think it could cause an unexpected outcome on the `if field not in exclude:` line below.